src/System.Private.CoreLib/shared/System/Runtime/Intrinsics/Vector128_1.cs

```diff
@@ -162,6 +165,34 @@ internal static void ThrowIfUnsupportedType()
         [CLSCompliant(false)]
         public Vector128<ulong> AsUInt64() => As<ulong>();
 
+        /// <summary>Determines whether the specified <see cref="Vector128{T}" /> is equal to the current instance.</summary>
+        /// <param name="other">The <see cref="Vector128{T}" /> to compare with the current instance.</param>
+        /// <returns><c>true</c> if <paramref name="other" /> is equal to the current instance; otherwise, <c>false</c>.</returns>
+        /// <exception cref="NotSupportedException">The type of the current instance (<typeparamref name="T" />) is not supported.</exception>
+        public bool Equals(Vector128<T> other)
+        {
+            ThrowIfUnsupportedType();
+
+            for (int i = 0; i < Count; i++)
+            {
+                if (!((IEquatable<T>)(GetElement(i))).Equals(other.GetElement(i)))
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        /// <summary>Determines whether the specified object is equal to the current instance.</summary>
+        /// <param name="obj">The object to compare with the current instance.</param>
+        /// <returns><c>true</c> if <paramref name="obj" /> is a <see cref="Vector128{T}" /> and is equal to the current instance; otherwise, <c>false</c>.</returns>
+        /// <exception cref="NotSupportedException">The type of the current instance (<typeparamref name="T" />) is not supported.</exception>
+        public override bool Equals(object obj)
+        {
+            return (obj is Vector128<T>) && Equals((Vector128<T>)(obj));
```

Comment:
Tanner curious why you didn't use the idiom `return (obj is Vector128<T> other) && Equals(other);`
Was it just an oversight or is there a specific reason.

Reply:
I've seen comments in other issues here that `is type name` produces poorer codegen than doing it manually

Reply:
> Was it just an oversight or is there a specific reason.

I wasn't thinking about it at the moment

Reply:
It tends to produce a a `box; dup` sequence that the jit can't optimize away, see eg #17670 and #14472.

Reply:
I didn't realize that pattern incurred a perf degradation. Thanks for the links.